ts/WoltLabSuite/Core/Ui/ItemList/LineBreakSeparatedText.ts

```diff
@@ -0,0 +1,251 @@
+/**
+ * UI element that shows individual lines of text as distinct list items but saves them as line
+ * break-separated text.
+ *
+ * @author  Matthias Schmidt
+ * @copyright  2001-2021 WoltLab GmbH
+ * @license  GNU Lesser General Public License <http://opensource.org/licenses/lgpl-license.php>
+ * @module  WoltLabSuite/Core/Ui/ItemList/LineBreakSeparatedText
+ * @since 5.4
+ */
+
+import * as UiConfirmation from "../Confirmation";
+import * as Language from "../../Language";
+import DomUtil from "../../Dom/Util";
+
+export interface LineBreakSeparatedTextOptions {
+  submitFieldName: string;
+}
+
+export class UiItemListLineBreakSeparatedText {
+  protected clearButton?: HTMLAnchorElement = undefined;
+  protected itemInput?: HTMLInputElement = undefined;
+  protected readonly itemList: HTMLUListElement;
+  protected items = new Set<string>();
+  protected readonly options: LineBreakSeparatedTextOptions;
+
+  constructor(itemList: HTMLUListElement, options: LineBreakSeparatedTextOptions) {
+    this.itemList = itemList;
+    this.options = options;
+
+    this.itemList.closest("form")!.addEventListener("submit", () => this.submit());
+
+    this.initValues();
+    this.buildUi();
+  }
+
+  /**
+   * Adds an item to the list after clicking on the "add" button.
+   */
+  protected addItem(event: Event): void {
+    event.preventDefault();
+
+    const item = this.itemInput!.value.trim();
```

Comment:
Please use something like `const itemInput = this.itemInput!;` to avoid sprinkling `!` all over the code.